hw-L06-reflection-test-runner/src/main/java/com/abrenchev/TestRunner.java

```diff
@@ -0,0 +1,87 @@
+package com.abrenchev;
+
+import com.abrenchev.annotations.After;
+import com.abrenchev.annotations.Before;
+import com.abrenchev.annotations.Test;
+import com.abrenchev.exceptions.TestRunnerException;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class TestRunner {
+    private Class<?> getClassByName(String fullClassName) {
+        try {
+            return Class.forName(fullClassName);
+        } catch (ClassNotFoundException exception) {
+            throw new TestRunnerException("Could not find class: " + fullClassName);
+        }
+    }
+
+    private Object createInstance(Class<?> clazz) {
+        try {
+            return clazz.getDeclaredConstructor().newInstance();
+        } catch (NoSuchMethodException | IllegalAccessException exception) {
+            throw new TestRunnerException("Provided class has no callable constructor");
```

Comment:
```suggestion
            throw new TestRunnerException("Provided class has no callable constructor", exception);
```
Всегда передавайте исходное исключение. Может пригодится, чтобы добраться до сути проблемы.